x-pack/plugins/rollup/public/crud_app/sections/job_create/steps/components/field_chooser.js

```diff
@@ -43,41 +60,77 @@ export class FieldChooser extends Component {
     });
   };
 
-  onSelect = (field) => {
-    this.props.onSelectField(field);
-  }
-
   render() {
     const {
       buttonLabel,
       columns,
       fields,
+      prompt,
+      onSelectField,
     } = this.props;
 
-    const { isOpen } = this.state;
+    const { isOpen, searchValue } = this.state;
 
-    const button = (
-      <EuiButton
-        iconType="arrowDown"
-        iconSide="right"
-        onClick={this.onButtonClick}
-      >
-        {buttonLabel}
-      </EuiButton>
-    );
+    const getRowProps = (item) => {
+      return {
+        onClick: () => {
+          onSelectField(item);
+        },
+      };
+    };
+
+    const searchedItems = searchValue ? fields.filter(item => (
+      item.name.toLowerCase().includes(searchValue.trim().toLowerCase())
+    )) : fields;
+
+    let flyout;
+
+    if (isOpen) {
+      flyout = (
+        <EuiFlyout
+          onClose={this.close}
+          aria-labelledby="fieldChooserFlyoutTitle"
+          size="m"
+          maxWidth={400}
+        >
+          <EuiFlyoutHeader>
+            <EuiTitle size="m" id="fieldChooserFlyoutTitle">
+              <h2>{buttonLabel}</h2>
+            </EuiTitle>
+
+            <EuiSpacer size="s" />
+
+            <EuiFieldSearch
+              placeholder={prompt}
+              value={searchValue}
+              onChange={this.onSearch}
+              aria-label={prompt}
+              fullWidth
+            />
+          </EuiFlyoutHeader>
+
+          <EuiFlyoutBody>
+            <EuiBasicTable
+              items={searchedItems}
```

Comment:
I'm able to get the flyout in a funky state where it shows fields I've already added and lets me add duplicates:

![image](https://user-images.githubusercontent.com/1965714/46195025-f7620500-c2b7-11e8-8a04-be92b53f39a0.png)

1. in Histogram, select bytes and memory
2. click Next, select phpmemory for Metrics
3. click back, open Histogram field selector - you will see that bytes and memory appear as options and can be added as duplicates

Reply:
Great catch, will fix